search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java

```diff
@@ -0,0 +1,300 @@
+    public static final String ADDITIONAL_IDENTIFIERS = "additionalIdentifiers";
+    public static final String SOURCE_NAME = "sourceName";
+    public static final String HANDLE_IDENTIFIER = "HandleIdentifier";
+    public static final String VALUE = "value";
+    public static final String SCOPUS_IDENTIFIER = "ScopusIdentifier";
+    public static final String CRISTIN_IDENTIFIER = "CristinIdentifier";
+    public static final String STATUS = "status";
+    public static final String CREATED_DATE = "createdDate";
+    public static final String MODIFIED_DATE = "modifiedDate";
+    public static final String PUBLISHED_DATE = "publishedDate";
+    public static final String NAME = "name";
+    public static final String DOI = "doi";
+    public static final String JOURNAL = "Journal";
+    public static final String PUBLISHER = "Publisher";
+    public static final String CONTRIBUTORS_PREVIEW = "contributorsPreview";
+    public static final String CORRESPONDING_AUTHOR = "correspondingAuthor";
+    public static final String IDENTITY = "identity";
+    public static final String SEQUENCE = "sequence";
+    public static final String ROLE = "role";
+    private final ObjectMapper objectMapper = dtoObjectMapper.copy();
+    public static final String ENTITY_DESCRIPTION = "entityDescription";
+    public static final String REFERENCE = "reference";
+    public static final String PUBLICATION_CONTEXT = "publicationContext";
+    public static final String SERIES = "series";
+
+    public SimplifiedResourceModelMutator() {
+        objectMapper.configure(SerializationFeature.WRITE_NULL_MAP_VALUES, false);
+        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+    }
+
+    public static String path(String... path) {
+        return Arrays.stream(path).collect(Collectors.joining("."));
+    }
+
+    public static List<String> getIncludedFields() {
+        return List.of(
+                ID,
+                path(ENTITY_DESCRIPTION, REFERENCE, PUBLICATION_INSTANCE, TYPE),
+                path(ENTITY_DESCRIPTION, REFERENCE, DOI),
+                path(
+                        ENTITY_DESCRIPTION,
+                        REFERENCE,
+                        PUBLICATION_CONTEXT), // TODO: Narrow down further?
+                path(DESCRIPTION),
+                path(ENTITY_DESCRIPTION, MAIN_TITLE),
+                path(ENTITY_DESCRIPTION, ABSTRACT),
+                path(ENTITY_DESCRIPTION, ALTERNATIVE_TITLES),
+                path(ENTITY_DESCRIPTION, CONTRIBUTORS_COUNT),
+                path(ENTITY_DESCRIPTION, CONTRIBUTORS_PREVIEW),
+                path(ENTITY_DESCRIPTION, PUBLICATION_DATE));
+    }
+
+    @Override
+    public JsonNode transform(JsonNode source) {
+        return (JsonNode)
+                attempt(() -> objectMapper.valueToTree(transformToDto(source))).orElseThrow();
+    }
+
+    private ResourceSearchResponse transformToDto(JsonNode source) throws IOException {
```

Comment:
I'd extract every possible method with a good name as this is reasonably complex.